ts/hooks/usePolling.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+
+export function usePolling<T>(fn: () => Promise<T>, delay: number) {
```

Comment:
I think you should rename delay to something like `pollInterval` or `repeatInterval`. Because the execution is run at least once on mount regardless of what value delay is.